### ISSUE TITLE
[a11y]: Modal: Passive modal appears to have keyboard trap

### ISSUE THREAD

**Issue by tay1orjones** [carbon-design-system/carbon#13713]
Originally outlined in #12727:

> This is a result of the fact that none of the modal examples except state manager are dismissable. On the others, it just results in an inability to dismiss. Here, it seems like a keyboard failure, since once the user tabs into the modal, they cannot get out or do ANYTHING. Why not allow all of them to be dismissable, even if you don't make all launchable (like the State manager).
